### PR TITLE
Cleans up streamer tests using the aggregator and mocked hive blocks db

### PR DESCRIPTION
**Pull request by mattrltrent**

### CODE REVIEW

File: modules/hive/streamer/streamer_test.go [hunk 1]
```diff
@@ -318,126 +258,44 @@ func TestFetchStoreBlocks(t *testing.T) {
...
-		_, err := s.Start().Await(context.Background())
-		assert.NoError(t, err)
-	}()
-	defer func() { assert.NoError(t, s.Stop()) }()
+	test_utils.RunPlugin(t, agg)
 
 	seenBlocks := make(map[int]int)
 
-	sr := streamer.NewStreamReader(hiveBlocks, func(block hive_blocks.HiveBlock) {
+	sr := streamer.NewStreamReader(mockHiveBlocks, func(block hive_blocks.HiveBlock) {
```
Comment (Geo25rey) [inline modules/hive/streamer/streamer_test.go:294]: `sr` should be part of aggregate

File: modules/hive/streamer/streamer_test.go [hunk 2]
```diff
@@ -821,7 +491,7 @@ func TestStreamReaderPauseResumeStop(t *testing.T) {
 	sr.Resume()
 
 	// seed
-	seedBlockData(t, hiveBlocks, 20)
+	seedBlockData(t, mockHiveBlocks, 20)
 
 	time.Sleep(1 * time.Second)
```
Comment (Geo25rey) [inline modules/hive/streamer/streamer_test.go:493]: Should make sure Stop() does not error

File: modules/hive/streamer/streamer_test.go [hunk 3]
```diff
@@ -887,13 +523,12 @@ func TestStreamPauseResumeStop(t *testing.T) {
...
-	s := streamer.NewStreamer(&MockBlockClient{}, hiveBlocks, []streamer.FilterFunc{filter}, nil)
+	s := streamer.NewStreamer(&MockBlockClient{}, mockHiveBlocks, []streamer.FilterFunc{filter}, nil)
 	assert.NoError(t, s.Init())
 	go func() {
 		_, err := s.Start().Await(context.Background())
 		assert.NoError(t, err)
 	}()
-	defer func() { assert.NoError(t, s.Stop()) }()
 
 	time.Sleep(3 * time.Second)
```
Comment (Geo25rey) [inline modules/hive/streamer/streamer_test.go:530]: Stop() error check here too

File: modules/hive/streamer/streamer_test.go [hunk 4]
```diff
@@ -923,141 +558,63 @@ func TestStreamPauseResumeStop(t *testing.T) {
...
 	}
 
-	s := streamer.NewStreamer(&MockBlockClient{}, hiveBlocks, []streamer.FilterFunc{filter1, filter2}, nil)
-	assert.NoError(t, s.Init())
-	go func() {
-		_, err := s.Start().Await(context.Background())
-		assert.NoError(t, err)
-	}()
-	defer func() { assert.NoError(t, s.Stop()) }()
+	s := streamer.NewStreamer(&MockBlockClient{}, mockHiveBlocks, []streamer.FilterFunc{filter1, filter2}, nil)
```
Comment (Geo25rey) [inline modules/hive/streamer/streamer_test.go:611]: This can be aggregated too

File: modules/hive/streamer/streamer_test.go [hunk 5]
```diff
@@ -1117,13 +641,9 @@ func TestBlockLag(t *testing.T) {
...
 	}
 
-	s := streamer.NewStreamer(&MockBlockClient{}, hiveBlocks, []streamer.FilterFunc{filter}, nil)
-	assert.NoError(t, s.Init())
-	assert.Equal(t, streamer.DefaultBlockStart, s.StartBlock())
-	go func() {
-		_, err := s.Start().Await(context.Background())
-		assert.NoError(t, err)
-	}()
+	s := streamer.NewStreamer(&MockBlockClient{}, mockHiveBlocks, []streamer.FilterFunc{filter}, nil)
```
Comment (Geo25rey) [inline modules/hive/streamer/streamer_test.go:639]: This can be aggregated

File: modules/hive/streamer/streamer_test.go [hunk 6]
```diff
@@ -1502,53 +840,26 @@ func TestNestedArrayStructure(t *testing.T) {
...
 	// compare the original and fetched blocks
 	//
 	// the reason we have to do this is because we store it internally in a different format so we
 	// want to ensure that our retrieval and conversion function is working correctly
-	assert.Equal(t, originalBlock, &fetchedBlocks[0])
+	assert.Equal(t, originalBlock, fetchedBlocks[0])
 }
 
 // todo: Vaultec's experiments
 func TestVaultecExperiments(t *testing.T) {
```
Comment (Geo25rey) [inline modules/hive/streamer/streamer_test.go:859]: @vaultec81 should this test be removed? It doesn't look like a unit test?

Comment (vaultec81) [reply modules/hive/streamer/streamer_test.go:859]: Yeah it can be removed at this point

**Comment by mattrltrent** [comment]
Unnecessary unit test removed as per Vaultec's suggestion and the rest of the tests cleaned up!

**Pull request merged**